### ISSUE TITLE
nss_resolve assertion failure in strv_find with systemd 247

### ISSUE THREAD

**Issue by tanriol** [systemd/systemd#17870]
**systemd version the issue has been seen with**
systemd 247 (247_rc2 and 247.1 versions in Gentoo, but not 247_rc1)

**System details**
 Gentoo Linux, amd64, 5.9.(various) kernels

**Problem description**

Firefox Nightly has rare crashes (once an hour or so) in `libnss_resolve.so.2`, which have the following error message
```
Assertion 'name' failed at src/basic/strv.c:22, function strv_find(). Aborting.
```

At the moment I don't have a stack trace for this assertion because Firefox fails to capture it.

**Comment by tanriol** [comment]
The next couple of frames, assuming I've read them correctly, implies that the error happens in `error_shall_fallback`, presumably because it's somehow called with NULL `error_id`.

**Comment by tanriol** [comment]
Caught it with `gdb`. Yes, the abort happens at [`nss-resolve.c:205`](https://github.com/systemd/systemd-stable/blob/v247-stable/src/nss-resolve/nss-resolve.c#L205).

**Comment by tanriol** [comment]
Also, `r` was -62, which seems to correspond to `-ETIME`.

**Comment by yuwata** [comment]
Thank you for the investigation. Fix is waiting in #17873.

**Comment by andrejpodzimek** [comment]
Is there a workaround for this or do we need to wait for the next `systemd` release?
